### ISSUE TITLE
Can't change default email templates

### ISSUE THREAD

**Issue by Luca8991** [supabase/supabase#7951]
# Bug report

## Describe the bug

Setting email templates from environment variables doesn't work in [self-hosted Docker Supabase](https://github.com/supabase/supabase/tree/master/docker)
In [self-hosted Docker Supabase](https://github.com/supabase/supabase/tree/master/docker), after setting `GOTRUE_MAILER_TEMPLATES_EMAIL_CHANGE` I'm still receiving the email with the [default template](https://github.com/supabase/gotrue/blob/master/mailer/template.go#L50-L54).

## To Reproduce

1. use [docker-comopse.yml](https://github.com/supabase/supabase/blob/master/docker/docker-compose.yml) to launch a self-hosted Supabase project
2. set `GOTRUE_MAILER_TEMPLATES_EMAIL_CHANGE` to (as per [docs](https://supabase.com/docs/reference/tools/reference-auth#email-templates)):
    ```
    GOTRUE_MAILER_TEMPLATES_EMAIL_CHANGE: "<h2>Confirm your Email</h2><p>Changing from {{ .Email }} to {{ .NewEmail }}:</p><p><a href='{{ .ConfirmationURL }}'>Change Email</a></p><p>Alternatively, enter the code: {{ .Token }}</p><p>Thanks</p>"
    ```
4. update user details changing email using [update()](https://supabase.com/docs/reference/javascript/auth-update) from Nodejs client (after authenticating user)
5. still receive change email confirmation with [default template](https://github.com/supabase/gotrue/blob/master/mailer/template.go#L50-L54):
    ```
    Confirm email address change
    Follow this link to confirm the update of your email address from [luca.berte98@hotmail.it](mailto:luca.berte98@hotmail.it) to [superluki31@gmail.com](mailto:superluki31@gmail.com):

    [Change email address link] (I won't paste it here)

    Alternatively, enter the code: <email-change-token>
    ```

**N.B.:** setting [custom subjects](https://supabase.com/docs/reference/tools/reference-auth#email) from environment variables works as expected. 

## Expected behavior

Receive the email change confirmation with the custom template set from environment variables.

## System information

- OS: Ubuntu 20.04 LTS
- Docker: Docker version 22.06.0-beta.0, build 3e9117b
- docker-compose: Docker Compose version v2.6.1
- supabase/gotrue: v2.5.21 (Docker image)
- Version of supabase-js: 1.35.4
- Version of Node.js: v16.14.1

**Comment by christianfl** [comment]
Hey! Funny, I recently wrote a short experience report on self hosting supabase and mentioned the exact same problem here: https://github.com/supabase/supabase/discussions/8004

I struggled, too - but it resolved itself. Look very carefully at the docs: The variable does not want plain html text, it needs a URL string which itself points to a an HTML template file. Try that and it works. Yes, it is a bit strange.

**Comment by AndrewTelkamp** [comment]
I'm experiencing this same issue for Template without self-hosting. No matter what I set for the template I get the generic email. E.g.

<img width="1470" alt="Screen Shot 2022-08-07 at 11 41 53 AM" src="https://user-images.githubusercontent.com/20894667/183304031-e449dbd3-4012-464e-bf80-0868954aa92b.png">


<img width="410" alt="Screen Shot 2022-08-07 at 11 39 14 AM" src="https://user-images.githubusercontent.com/20894667/183303931-19ebc15f-467d-4fca-85a7-0562a3e67eac.png">

I've clicked save and refreshed to ensure the changes were made.  These are my Auth Settings if it helps w/ troubleshooting

<img width="1118" alt="Screen Shot 2022-08-07 at 11 44 12 AM" src="https://user-images.githubusercontent.com/20894667/183304133-a08d5126-7e96-4ce0-8bed-f2be5d6be7f3.png">

**Comment by christianfl** [comment]
@AndrewTelkamp It's the same. U need to pass a URL and not html.

**Comment by AndrewTelkamp** [comment]
I don't think this is the case @christianfl. I was definitely messing up the URL (needed to remove the brackets), but I am able to get inline HTML to work fine. For example, this works fine, and I'm receiving an updated template now.
